protocols/autonat/src/protocol.rs

```diff
@@ -269,28 +266,30 @@ impl DialResponse {
 
     pub fn into_bytes(self) -> Vec<u8> {
         let dial_response = match self.result {
-            Ok(addr) => structs_proto::message::DialResponse {
-                status: Some(0),
-                status_text: self.status_text,
-                addr: Some(addr.to_vec()),
+            Ok(addr) => {
+                let mut res = structs_proto::message::DialResponse::new();
+                res.set_status(structs_proto::message::ResponseStatus::OK);
+                res.set_addr(addr.to_vec());
+                res.statusText = self.status_text;
+
+                res
             },
-            Err(error) => structs_proto::message::DialResponse {
-                status: Some(error.into()),
-                status_text: self.status_text,
-                addr: None,
+            Err(error) => {
+                let mut res = structs_proto::message::DialResponse::new();
+                res.clear_addr();
+                res.set_status(structs_proto::message::ResponseStatus::from_i32(error.into()).expect("Valid ResponseError."));
+                res.statusText = self.status_text;
+
+                res
             },
         };
 
-        let msg = structs_proto::Message {
-            r#type: Some(structs_proto::message::MessageType::DialResponse as _),
-            dial: None,
-            dial_response: Some(dial_response),
-        };
+        let mut msg = structs_proto::Message::new();
+        msg.set_type(structs_proto::message::MessageType::DIAL_RESPONSE as _);
+        msg.dial = MessageField::none();
+        msg.dialResponse = MessageField::some(dial_response);
```

Comment:
Well this is kind of an annoying API but I guess that is what you get with generated code.

Reply:
> Well this is kind of an annoying API

Agreed. It would be nice if it generated setters for `MessageField`s.